api/metrics/v1/http.go

```diff
@@ -153,6 +153,7 @@ func NewHandler(read, write, rulesEndpoint *url.URL, upstreamCA []byte, upstream
 	r.Use(func(handler http.Handler) http.Handler {
 		return c.instrument.NewHandler(nil, handler)
 	})
+	r.Use(server.StripTenantPrefix("/api/metrics/v1"))
```

Comment:
if this happens here, any middleware which expects the original path to be untouched (rate limiter for example) will be broken. Hence why it was applied directly after the middleware in each case

Reply:
Hummm, I wasn't sure about it. I will spend some time thinking hard about the need for this `StripTenantPrefix` thing. I want to get rid of it, because modifying the path like this is kinda confusing and annoying for all the subsequent middlewares.

Reply:
it i there because without it, we would need to add a handler for each path for each tenant no?

Reply:
+1 to what @PhilipGough mentioned